tests/ProjectStorageBackendTest.php

```diff
@@ -0,0 +1,149 @@
+
+        $this->assertArrayHasKey($backendName, $backends);
+        $this->assertEquals($backendName, $project['defaultBackend']);
+
+        $this->assertEquals($backendToAssign['id'], $backends[$backendName]['id']);
+
+        // let's try to create a bucket in project now
+
+        $token = $this->client->createProjectStorageToken($project['id'], [
+            'description' => 'test',
+            'expiresIn' => 60,
+            'canManageBuckets' => true,
+        ]);
+
+        $sapiClient = new \Keboola\StorageApi\Client([
+            'url' => getenv('KBC_MANAGE_API_URL'),
+            'token' => $token['token'],
+        ]);
+        $bucketId = $sapiClient->createBucket('test', 'in');
+        $bucket = $sapiClient->getBucket($bucketId);
+        $this->assertEquals($backendName, $bucket['backend']);
+
+        $sapiClient->dropBucket($bucketId);
+
+        $this->client->removeProjectStorageBackend($project['id'], $backendToAssign['id']);
+
+        $this->client->deleteProject($project['id']);
+        $this->client->purgeDeletedProject($project['id']);
+    }
+
+    public function testStorageBackendRemove()
+    {
+        $name = 'My org';
+        $organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => $name,
+        ]);
+
+        $project = $this->client->createProject($organization['id'], [
+            'name' => 'My test',
+        ]);
+
+        $this->assertArrayHasKey('backends', $project);
+        $this->assertEquals('snowflake', $project['defaultBackend']);
+        $this->assertCount(1, $project['backends']);
+
+        $this->client->removeProjectStorageBackend($project['id'], reset($project['backends'])['id']);
+
+        $project = $this->client->getProject($project['id']);
+        $this->assertEmpty($project['backends']);
+    }
+
+    public function testStorageBackendShouldNotBeRemovedIfThereAreBuckets()
+    {
+        $name = 'My org';
+        $organization = $this->client->createOrganization($this->testMaintainerId, [
+            'name' => $name,
+        ]);
+
+        $project = $this->client->createProject($organization['id'], [
+            'name' => 'My test',
+        ]);
+
+        $token = $this->client->createProjectStorageToken($project['id'], [
+            'description' => 'test',
+            'expiresIn' => 60,
+            'canManageBuckets' => true,
+        ]);
+
+        $sapiClient = new \Keboola\StorageApi\Client([
+            'url' => getenv('KBC_MANAGE_API_URL'),
+            'token' => $token['token'],
+        ]);
+        $sapiClient->createBucket('test', 'in');
+
+        try {
+            $this->client->removeProjectStorageBackend($project['id'], reset($project['backends'])['id']);
+            $this->fail('Backend should not be removed');
+        } catch (ClientException $e) {
+            $this->assertEquals(400, $e->getCode());
+            $this->assertEquals('bucketsExists', $e->getStringCode());
```

Comment:
```suggestion
            $this->assertEquals('bucketsExist', $e->getStringCode());
```

Buď "bucket exists" (3. osoba jednotné číslo) nebo "buckets exist" (množné číslo)